test/unit/util_new-block-ids.js

```diff
@@ -0,0 +1,64 @@
+const newBlockIds = require('../../src/util/new-block-ids');
+const simpleStack = require('../fixtures/simple-stack');
+const tap = require('tap');
+const test = tap.test;
+
+let originals;
+let newBlocks;
+
+tap.beforeEach(done => {
+    originals = simpleStack;
+    // Will be mutated so make a copy first
+    newBlocks = JSON.parse(JSON.stringify(simpleStack));
+    newBlockIds(newBlocks);
+    done();
```

Comment:
In a separate test I've had the following structure, we should probably decide on one or the other (it doesn't necessarily have to happen now):

```
tap.beforeEach(() => {
    ...
    return Promise.resolve(null);
});
```